script.js

```diff
@@ -1,4 +1,265 @@
+        cardName = "jack";
+      } else if (cardName == 12) {
+        cardName = "queen";
+      } else if (cardName == 13) {
+        cardName = "king";
+      }
+
+      // Create a new card with the current name, suit, and rank
+      var card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+      };
+
+      // Add the new card to the deck
+      cardDeck.push(card);
+
+      // Increment rankCounter to iterate over the next rank
+      rankCounter += 1;
+    }
+
+    // Increment the suit index to iterate over the next suit
+    suitIndex += 1;
+  }
+
+  // Return the completed card deck
+  return cardDeck;
+};
+
+// Shuffle the deck and save it in a new variable shuffledDeck
+// to communicate that we have shuffled the deck.
+// console.log(shuffledDeck);
+var shuffledDeck = shuffleCards(makeDeck());
+
+// var playerDraw = function (){
+
+// var firstCard =
+
+// }
+
+//Card calculator
+var calculateScore = function (drawnDeck) {
+  var score = 0;
+
+  console.log(drawnDeck);
+  score += drawnDeck[0].rank;
+  for (var i = 1; i < drawnDeck.length; i++) {
+    //Convert rank of Ace to 11 if less than 10
+    if (score < 10 && drawnDeck[i].name == "ace") {
+      console.log("ace");
+      drawnDeck[i].rank = 11;
+    }
+    score += drawnDeck[i].rank;
+    console.log(score);
+  }
+
+  return score;
+};
+
+//Ace value
+
+// var checkForAce = function (drawnDeck) {
+//   var score = 0;
+
+//   for (i = 0; i < drawnDeck.length; i++) {
+//     if (score > 10 && drawnDeck[i].name == "ace") {
+//       drawnDeck[i].rank = 1;
+//     }
+//     score += drawnDeck[i].rank;
+//   }
+//   console.log(score);
+//   return score;
+// };
+
+//cardDrawn Statement
+
+var cardDrawnsStatement = function (person, drawnDeck) {
+  var statement = `${person} has drawn`;
+  console.log(statement);
+  for (i = 0; i < drawnDeck.length; i++) {
+    if (i != drawnDeck.length - 1) {
+      statement += ` ${drawnDeck[i].name} of ${drawnDeck[i].suit} and`;
+    } else statement += ` ${drawnDeck[i].name} of ${drawnDeck[i].suit}.`;
+  }
+  console.log(statement);
+  return statement;
+};
+
+//convert Jack, Queen, King to rank 10
+var convertValues = function (drawnDeck) {
+  for (i = 0; i < drawnDeck.length; i++) {
+    if (
+      drawnDeck[i].name == "king" ||
+      drawnDeck[i].name == "queen" ||
+      drawnDeck[i].name == "jack"
+    ) {
+      drawnDeck[i].rank = 10;
+    }
+  }
+  return drawnDeck;
+};
+
+//Sort the rank according to order and put Ace at the back
+var selectionSort = function (drawnCards) {
+  //Sort the score according to DESC order
+
+  for (var i = 0; i < drawnCards.length; i++) {
+    //set min to the current iteration of i
+    var min = i;
+    for (var j = i + 1; j < drawnCards.length; j++) {
+      if (drawnCards[j].rank > drawnCards[min].rank) {
+        min = j;
+      }
+    }
+    var temp = drawnCards[i];
+    drawnCards[i] = drawnCards[min];
+    drawnCards[min] = temp;
+  }
```

Comment:
👍. though should min be named as max instead? because j card is always the bigger card from linke 164 to 165